manifests/module/detail.pp

```diff
@@ -9,12 +9,8 @@
   String $header                                   = '%t',
   Optional[Freeradius::Boolean] $locking           = undef,
   Optional[Freeradius::Boolean] $log_packet_header = undef,
-  Array[String] $suppress                          = [],
+  Optional[Array[String]] $suppress                = [],
```

Comment:
This Optional here has no effect - Optional means "can be undef, or ..." - but the default is `[]` so it is never undef.

The rest of the code here needs a tidy up where the defaults are not undef :)